File: app/src/main/java/com/galtashma/parsedashboard/adapters/ParseObjectFieldsAdapter.java
package com.galtashma.parsedashboard.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import androidx.annotation.NonNull;

import com.galtashma.parsedashboard.ParseField;
import com.galtashma.parsedashboard.R;
import com.lucasurbas.listitemview.ListItemView;

import java.util.List;
import java.util.Map;

/**
 * Created by gal on 3/16/18.
 */

public class ParseObjectFieldsAdapter extends ArrayAdapter<ParseField> {

    private View.OnLongClickListener longClickListener = null;

    public ParseObjectFieldsAdapter(@NonNull Context context, @NonNull List<ParseField> objects) {
        super(context, R.layout.list_item, objects);
    }

    @NonNull
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {
        ParseField details = getItem(position);
        if(convertView == null) {
            convertView = LayoutInflater.from(this.getContext()).inflate(R.layout.list_item, parent, false);
        }

        ListItemView view = (ListItemView) convertView;

        if (details == null){
            return convertView;
        }

        if (details.value == null || details.value.isEmpty()){
            view.setTitle("<empty>");
        } else {
            view.setTitle(details.value);
        }

        view.setSubtitle(details.key);
        view.setOnLongClickListener(longClickListener);

        return convertView;
    }

    public void setLongClickListener(View.OnLongClickListener longClickListener) {
        this.longClickListener = longClickListener;
    }
}

File: app/src/main/java/com/galtashma/parsedashboard/adapters/ParseObjectFieldsAdapter.kt
package com.galtashma.parsedashboard.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import androidx.annotation.NonNull
import com.galtashma.parsedashboard.ParseField
import com.galtashma.parsedashboard.R
import com.lucasurbas.listitemview.ListItemView

/**
 * Created by gal on 3/16/18, rewritten by Cyb3rKo on 05/12/22.
 */

class ParseObjectFieldsAdapter(
    @NonNull context: Context,
    @NonNull objects: List<ParseField>
) : ArrayAdapter<ParseField>(context, R.layout.list_item, objects) {

    private var longClickListener: View.OnLongClickListener? = null

    override fun getView(
        position: Int,
        convertView: View?,
        parent: ViewGroup
    ): View {
        val details = getItem(position)

        var finalView = convertView
        if (convertView == null) {
            finalView = LayoutInflater.from(context).inflate(R.layout.list_item, parent, false)
        }

        val view = finalView as ListItemView

        if (details == null) {
            return finalView
        }

        view.title = details.value.ifBlank { "<empty>" }

        view.subtitle = details.key
        view.setOnLongClickListener(longClickListener)

        return finalView
    }

    fun setLongClickListener(longClickListener: View.OnLongClickListener) {
        this.longClickListener = longClickListener
    }
}
